shortfin/python/shortfin_apps/llm/components/service.py

```diff
@@ -113,9 +119,9 @@ def initialize_worker_and_fiber(self):
 
         self.devices = fibers[0].devices_dict.values()
         self.main_worker = self.sysman.ls.create_worker(f"{self.name}-inference")
-        self.main_fiber = self.sysman.ls.create_fiber(self.main_worker)
-        self.prefill_fiber = self.sysman.ls.create_fiber(self.main_worker)
-        self.decode_fiber = self.sysman.ls.create_fiber(self.main_worker)
+        self.main_fiber = self.fiber_pool.fibers[0]
+        self.prefill_fiber = self.fiber_pool.fibers[0]
+        self.decode_fiber = self.fiber_pool.fibers[0]
```

Comment:
This assigns the same fiber to `main`, `prefill`, and `decode` where previously they were on different fibers. I had a discussion with Stephen and there is a high likelihood that me want to revert the fiber pool work as its current implementation actually slows down processing.